ddtrace/contrib/algoliasearch/patch.py

```diff
@@ -0,0 +1,138 @@
+    _w(algoliasearch.index, 'Index.search', _get_patched_search(algoliasearch))
+    Pin(
+        service=config.algoliasearch.service_name, app=config.algoliasearch.app,
+        app_type=config.algoliasearch.app_type
+    ).onto(algoliasearch.index.Index)
+
+
+def unpatch():
+    try:
+        algoliasearch = import_module('algoliasearch')
+    except ImportError:
+        return
+
+    if getattr(algoliasearch, DD_PATCH_ATTR, False):
+        setattr(algoliasearch, DD_PATCH_ATTR, False)
+        _u(algoliasearch.index.Index, 'search')
+
+
+# DEV: this maps serves the dual purpose of enumerating the algoliasearch.search() query_args that
+# will be sent along as tags, as well as converting arguments names into tag names compliant with
+# tag naming recommendations set out here: https://docs.datadoghq.com/tagging/
+QUERY_ARGS_DD_TAG_MAP = {
+    'page': 'page',
+    'hitsPerPage': 'hits_per_page',
+    'attributesToRetrieve': 'attributes_to_retrieve',
+    'attributesToHighlight': 'attributes_to_highlight',
+    'attributesToSnippet': 'attributes_to_snippet',
+    'minWordSizefor1Typo': 'min_word_size_for_1_typo',
+    'minWordSizefor2Typos': 'min_word_size_for_2_typos',
+    'getRankingInfo': 'get_ranking_info',
+    'aroundLatLng': 'around_lat_lng',
+    'numericFilters': 'numeric_filters',
+    'tagFilters': 'tag_filters',
+    'queryType': 'query_type',
+    'optionalWords': 'optional_words',
+    'distinct': 'distinct'
+}
+
+
+def _get_patched_search(algoliasearch):
+    def _search(func, instance, wrapt_args, kwargs):
+        """
+            wrapt_args is called the way it is to distinguish it from the 'args'
+            argument to the algoliasearch.index.Index.search() method.
+        """
+        pin = Pin.get_from(instance)
+        if not pin or not pin.enabled():
+            return func(*wrapt_args, **kwargs)
+
+        with pin.tracer.trace('algoliasearch.search') as span:
+            if not span.sampled:
+                return func(*wrapt_args, **kwargs)
+
+            span.service = pin.service
+            span.span_type = SEARCH_SPAN_TYPE
+            if config.algoliasearch.collect_query_text:
+                span.set_tag('query_text', kwargs.get('query', wrapt_args[0]))
+
+            query_args = kwargs.get('args', wrapt_args[1] if len(wrapt_args) > 1 else None)
+            if query_args is None:
+                # try 'searchParameters' as the name, which seems to be a deprecated argument name
+                # that is still in use in the documentation but not in the latest algoliasearch
+                # library
+                query_args = kwargs.get('searchParameters', None)
+
+            if query_args and isinstance(query_args, dict):
+                for query_arg, tag_name in QUERY_ARGS_DD_TAG_MAP.items():
+                    value = query_args.get(query_arg, None)
+                    if value is not None:
+                        try:
+                            value = stringify(value)
```

Comment:
`stringify(value)` is actually handled by `span.set_tag` so it is not necessary here.